File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/hub/FlutterModelSyncedEvent.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore.types.hub

class FlutterModelSyncedEvent(
    override val eventName: String,
    val model: String,
    private val isFullSync: Boolean,
    private val isDeltaSync: Boolean,
    private val added: Int,
    private val updated: Int,
    val deleted: Int
) : FlutterHubEvent {
    override fun toValueMap(): Map<String, Any> {
        return mapOf(
            "eventName" to this.eventName,
            "model" to this.model,
            "isFullSync" to this.isFullSync,
            "isDeltaSync" to this.isDeltaSync,
            "added" to this.added,
            "updated" to this.updated,
            "deleted" to this.deleted
        )
    }
}

File: packages/amplify_datastore/ios/Classes/types/hub/FlutterModelSyncedEvent.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Foundation
import Amplify
import AmplifyPlugins

struct FlutterModelSyncedEvent: FlutterHubEvent {
    var eventName: String
    var modelName: String
    var isFullSync: Bool
    var isDeltaSync: Bool
    var added: Int
    var updated: Int
    var deleted: Int

    init(payload: HubPayload) throws {
        guard let modelSynced = payload.data as? ModelSyncedEvent else {
                  throw FlutterDataStoreError.hubEventCast
              }
        self.eventName = shortEventName(eventName: payload.eventName)
        self.modelName = modelSynced.modelName
        self.isFullSync = modelSynced.isFullSync
        self.isDeltaSync = modelSynced.isDeltaSync
        self.added = modelSynced.added
        self.updated = modelSynced.updated
        self.deleted = modelSynced.deleted
    }

    func toValueMap() -> [String: Any] {
        return [
            "eventName": eventName,
            "model": modelName,
            "isFullSync": isFullSync,
            "isDeltaSync": isDeltaSync,
            "added": added,
            "updated": updated,
            "deleted": deleted
        ]
    }
}
